src/bench/wallet_create_tx.cpp

```diff
@@ -0,0 +1,142 @@
+};
+
+TipBlock getTip(const CChainParams& params, const node::NodeContext& context)
+{
+    auto tip = WITH_LOCK(::cs_main, return context.chainman->ActiveTip());
+    return (tip) ? TipBlock{tip->GetBlockHash(), tip->GetBlockTime(), tip->nHeight} :
+           TipBlock{params.GenesisBlock().GetHash(), params.GenesisBlock().GetBlockTime(), 0};
+}
+
+void generateFakeBlock(const CChainParams& params,
+                       const node::NodeContext& context,
+                       CWallet& wallet,
+                       const CScript& coinbase_out_script)
+{
+    TipBlock tip{getTip(params, context)};
+
+    // Create block
+    CBlock block;
+    CMutableTransaction coinbase_tx;
+    coinbase_tx.vin.resize(1);
+    coinbase_tx.vin[0].prevout.SetNull();
+    coinbase_tx.vout.resize(2);
+    coinbase_tx.vout[0].scriptPubKey = coinbase_out_script;
+    coinbase_tx.vout[0].nValue = 49 * COIN;
+    coinbase_tx.vin[0].scriptSig = CScript() << ++tip.tip_height << OP_0;
+    coinbase_tx.vout[1].scriptPubKey = coinbase_out_script; // extra output
+    coinbase_tx.vout[1].nValue = 1 * COIN;
+    block.vtx = {MakeTransactionRef(std::move(coinbase_tx))};
+
+    block.nVersion = VERSIONBITS_LAST_OLD_BLOCK_VERSION;
+    block.hashPrevBlock = tip.prev_block_hash;
+    block.hashMerkleRoot = BlockMerkleRoot(block);
+    block.nTime = ++tip.prev_block_time;
+    block.nBits = params.GenesisBlock().nBits;
+    block.nNonce = 0;
+
+    {
+        LOCK(::cs_main);
+        // Add it to the index
+        CBlockIndex* pindex{context.chainman->m_blockman.AddToBlockIndex(block, context.chainman->m_best_header)};
+        // add it to the chain
+        context.chainman->ActiveChain().SetTip(*pindex);
+    }
+
+    // notify wallet
+    const auto& pindex = WITH_LOCK(::cs_main, return context.chainman->ActiveChain().Tip());
+    wallet.blockConnected(kernel::MakeBlockInfo(pindex, &block));
+}
+
+struct PreSelectInputs {
+    // How many coins from the wallet the process should select
+    int num_of_internal_inputs;
+    // future: this could have external inputs as well.
+};
+
+static void WalletCreateTx(benchmark::Bench& bench, const OutputType output_type, bool allow_other_inputs, std::optional<PreSelectInputs> preset_inputs)
+{
+    const auto test_setup = MakeNoLogFileContext<const TestingSetup>();
+
+    CWallet wallet{test_setup->m_node.chain.get(), "", gArgs, CreateMockWalletDatabase()};
+    {
+        LOCK(wallet.cs_wallet);
+        wallet.SetWalletFlag(WALLET_FLAG_DESCRIPTORS);
+        wallet.SetupDescriptorScriptPubKeyMans();
+        if (wallet.LoadWallet() != DBErrors::LOAD_OK) assert(false);
+    }
+
+    // Generate destinations
+    CScript dest = GetScriptForDestination(getNewDestination(wallet, output_type));
+
+    // Generate chain; each coinbase will have two outputs to fill-up the wallet
+    const auto& params = Params();
+    unsigned int chain_size = 150000; // 150k blocks means 300k UTXO for the wallet (minus 200 due COINBASE_MATURITY)
```

Comment:
In 2d9ea15fe9773c3efe349579413b4a4e56c30454 "bench: benchmark transaction creation process"

150000 blocks is excessive. This makes the benchmark extremely slow to run. It increases the runtime of `make check` by several orders of magnitude. A few hundred blocks should be enough to illustrate the performance difference, I don't think we need more than a thousand, let one 150,000.

Reply:
I initially set it low, at 2k (4k utxo), but it wasn't really illustrative. This becomes a problem when your wallet has a high tx volume. Otherwise, traverse 4k utxo is quite fast, almost imperceptible.
e.g.
### PR
|               ns/op |                op/s |    err% |     total | benchmark
|--------------------:|--------------------:|--------:|----------:|:----------
|        1,427,966.60 |              700.30 |    1.2% |      0.08 | `WalletCreateTxUseOnlyPresetInputs`

### Master
|               ns/op |                op/s |    err% |     total | benchmark
|--------------------:|--------------------:|--------:|----------:|:----------
|        1,576,250.00 |              634.42 |    0.5% |      0.09 | `WalletCreateTxUseOnlyPresetInputs`
----

Let me try with 20k instead. We should try to get a middle-ground between the high value (which I agree that it's too high) and a number that can provide meaningful information on the context that this becomes a real problem for the user.

Reply:
20k is still a lot and the setup does still take quite a long time. While the real time difference may not be that noticeable, I think `ns/op` and `op/s`

Reply:
I'm not so convinced there. I do understand the `make check` time concern but that is limiting our testing capabilities. Some big issues aren't problems at all in a small scale.

As time isn't constant between runs and the difference is so small with 2000 blocks, stuff like this can easily go unnoticed for a really long time. Which kinda break the purpose of having a benchmark for it.

E.g. lets say that we have this benchmark in the code with the 2000 blocks param. And someone introduces a similar problematic (traverse the whole wallet map and do other stuff for no reason).
Would you care about it if the benchmark returns a time diff between 60-100 ns? or would you just say "it's ok"?.
We could be hurting badly big wallets with our changes without noticing it at all.

Saying that, will decrease the number to 10k blocks. So we get closer to each other number and can move forward with it.
I think that the 2k blocks number is too small to be helpful on this context.

----------------------

As a future thought, would be nice to be able to re-use setups in the benchmarks. So every `WalletCreateTx*` would not require to create a new chain per run.

Reply:
Even with 2000 blocks I am still able to measure a difference. It's also a lot more noticeable (and everything is a lot slower) if configured with `--enable-debug`.

Reply:
Interesting, that is why we have been seeing different pictures of this. Without the debug mode, the difference that have been seeing locally is almost imperceptible with 2k (less than 100 ns).
That explains why the CI was timing out before, benchmarks there are running on debug mode as well.